base/namedtuple.jl

```diff
@@ -103,6 +103,8 @@ firstindex(t::NamedTuple) = 1
 lastindex(t::NamedTuple) = nfields(t)
 getindex(t::NamedTuple, i::Int) = getfield(t, i)
 getindex(t::NamedTuple, i::Symbol) = getfield(t, i)
+getindex(t::NamedTuple, v::AbstractVector{<:Integer}) = (; map(i->keys(t)[i] => t[i], v)...)
+getindex(t::NamedTuple, v::AbstractVector{Symbol}) = (; map(i->i => t[i], v)...)
```

Comment:
We typically don't like to dispatch behaviors on the `eltype` of a Vector in Base (except for Bool). These functions should probably give the same result regardless of whether it gets passed a `Vector{Any}` or a `Vector{Union{Int, Symbol}}`:
```
getindex(t::NamedTuple, v::AbstractVector) = (; map(v) do i
          k = v isa Symbol ? v : keys(t)[i]
          return k => t[i]
     end...)
```